evidence/pool.go

```diff
@@ -300,8 +300,7 @@ func (evpool *Pool) removePendingEvidence(evidence types.Evidence) {
 }
 
 // listEvidence lists up to maxNum pieces of evidence for the given prefix key.
-// It is wrapped by PriorityEvidence and PendingEvidence for convenience.
-// If maxNum is -1, there's no cap on the size of returned evidence.
+// If maxNum is 0, there's no cap on the size of returned evidence.
```

Comment:
let's revert to using `-1` and make maxNum an int. what do you think? similar to CListMempool#ReapMaxTxs

Reply:
On a second thought, we should create a second function (listAllEvidence, AllPendingEvidence) and use it in tests. That's how we get flexibility in tests and safety in production.

Reply:
Okay yeah I can do that